python/cugraph/cugraph/dask/sampling/uniform_neighbor_sample.py

```diff
@@ -153,39 +121,45 @@ def uniform_neighbor_sample(input_graph,
         raise TypeError("fanout_vals must be a list, "
                         f"got: {type(fanout_vals)}")
 
-    ddf = input_graph.edgelist.edgelist_df
-    src_col_name = input_graph.renumber_map.renumbered_src_col_name
-    dst_col_name = input_graph.renumber_map.renumbered_dst_col_name
-
-    weight_t = ddf["value"].dtype
-    if weight_t == "int32":
-        ddf = ddf.astype({"value": "float32"})
-    elif weight_t == "int64":
-        ddf = ddf.astype({"value": "float64"})
-
-    num_edges = len(ddf)
-    data = get_distributed_data(ddf)
+    weight_t = input_graph.edgelist.edgelist_df["value"].dtype
 
     # start_list uses "external" vertex IDs, but if the graph has been
     # renumbered, the start vertex IDs must also be renumbered.
     if input_graph.renumbered:
         start_list = input_graph.lookup_internal_vertex_id(
             start_list).compute()
-    # FIXME: should we add this parameter as an option?
-    do_expensive_check = False
-
-    result = [client.submit(call_nbr_sampling,
-                            Comms.get_session_id(),
-                            wf[1],
-                            src_col_name,
-                            dst_col_name,
-                            num_edges,
-                            do_expensive_check,
-                            start_list,
-                            fanout_vals,
-                            with_replacement,
-                            workers=[wf[0]])
-              for idx, wf in enumerate(data.worker_to_parts.items())]
+
+    '''
+    start_list = dask_cudf.from_cudf(
+        start_list,
+        npartitions=input_graph.npartitions
+    )
+    start_list = get_distributed_data(start_list)
+    wait(start_list)
+    '''
+    import cupy
+    dummy = dask_cudf.from_cudf(
+        cudf.Series(cupy.ones(input_graph.npartitions)),
+        npartitions=input_graph.npartitions
+    )
+    dummy = get_distributed_data(dummy)
+
+    client = input_graph._client
+
+    result = [
+        client.submit(
+            _call_plc_uniform_neighbor_sample,
+            Comms.get_session_id(),
+            mg_graph,
+            start_list,
+            fanout_vals,
+            with_replacement,
+            workers=[w],
+        )
+        for mg_graph, (w, _) in zip(
+            input_graph._plc_graph, dummy.worker_to_parts.items()
```

Comment:
if the sole goal of `dummy` is just to get the workers addresses, you can just call `Comms.get_workers()` because that's what `get_distributed_data` does under the hood

Reply:
May be I am overthinking but I am worried the worker initially carrying a partition might not be the one we are passing in `line 157` regardless if we get the worker addresses from `dummy` or `Comms.get_workers()`: The old implementation ensured it doesn't happen. This will still work but it might cause some performance regression because it defeat the purpose of data locality.

Reply:
I'll make the change to Comms.get_workers().

I thought the API was consistent with the data being partitioned the same way each time.  But if it isn't I can store the workers along with each _plc graph, right?

Reply:
> I thought the API was consistent with the data being partitioned the same way each time
You are right. i just took a look and the API consistently [persist](https://github.com/rapidsai/cugraph/blob/branch-22.08/python/cugraph/cugraph/dask/common/part_utils.py#L93) the data been partitioned. So each worker will persist the data in the order they appear in Comms.get_workers.

Reply:
I figured out the correct way to do this and I'm fixing it now.

Reply:
Update: who_has doesn't seem to work right, but Comms.get_workers does.  I'm not sure why this is the case.

Reply:
That's weird. Did you copy the script I shared above?

Reply:
I did.

Reply:
Got it. I will take a look at tit tomorrow